powermon/__init__.py

```diff
@@ -142,89 +148,53 @@ def main():
         print("# default location ./powermon.yaml")
         print(yaml.dump(config))
         return
+    # debug config
+    log.info("config: %s", config)
 
-    # debug dump config
-    log.info(f"config: {config}")
+    # build mqtt broker object (optional)
+    mqtt_broker = MqttBroker(config=config.get("mqttbroker", {}))
+    log.debug("mqtt_broker: %s", mqtt_broker)
 
-    # Build mqtt broker
-    mqttconfig = config.get("mqttbroker", {})
-    mqtt_broker = MqttBroker(config=mqttconfig)
     # is this just a call to send an adhoc command to the queue?
+    # TODO: sort the use of powermon command line to send adhoc command
     if args.adhoc:
         print("ADHOC is todo")
         return
 
-    # debug dump mqttbroker details
-    log.debug(f"mqtt_broker: {mqtt_broker}")
-
-    # get the device config
-    # this is required, and contains:
-    #     device:
-    #       name:
-    #       id:
-    #       model:
-    #       manufacturer:
-    #       port:
-    #       commands:
-    device_config = config.get("device", None)
-    if not device_config:
-        log.error(f"No device definition in config. Check {args.configFile}?")
-        exit(1)
-
-    
-
-    # configure the daemon (optional)
-    #     daemon:
-    #       type: systemd  # noqa:
-    #       keepalive: 10
+    # build device object (required)
+    device = Device(config=config.get("device", None))
+    log.debug("device: %s", device)
+
+    # build the daemon object (optional)
     daemon = Daemon(config=config)
-    log.debug(f"daemon: {daemon}")
-
-    # config the port (required)
-    # config depends on port type
-    #   port:
-    #     baud: 2400
-    #     path: /dev/ttyUSB0
-    #     type: serial  # noqa:
-    #     protocol: PI30MAX
-    log.debug(f"deviceconfig: {device_config}")
-
-    
-    device = Device.fromConfig(device_config)
-    log.debug(f"device: {device}")
-    # error out if unable to configure port
-    if not device:
-        log.error(f"No config '{device_config}' found")
-        raise ConfigError(f"No port for config '{device_config}' found")
-
-    
+    log.debug("daemon: %s", daemon)
 
     # Get scheduled commands
     scheduling_config = config.get("scheduling", None)
-    log.debug(f"scheduling_config: {scheduling_config}")
+    log.debug("scheduling_config: %s", scheduling_config)
     schedule = Schedule.parseScheduleConfig(scheduling_config, device, mqtt_broker)
 
     log.debug(schedule)
 
     # initialize daemon
     daemon.initialize()
-    
+
     # Main working loop
-    keepLooping = True
+    keep_looping = True
     try:
         schedule.beforeLoop()
-        while keepLooping:
+        while keep_looping:
```

Comment:
while schedule.run_loop ?
or while true and schedule throws an exception when done??

Reply:
I had it as while schedule.run_loop at first and you can change it back to that if it's more readable. I thought there might be a different condition that needs to change the loop in the future. I don't know what it would be yet though

Reply:
We can also have exceptions (probably custom ones) to break the loop early and and use the catch blocks for dealing with these situations